examples/common/fdiff.py

```diff
@@ -38,26 +36,31 @@ def batch_end(self, state: State, logger: Logger):
 
             self.train_prev_loss = state.loss.item()
 
-            if self.train_prev_metric:
-                for k in state.train_metric_values.keys():
-                    logger.log_metrics({
-                        f'metrics/train/{k}_fdiff':
-                            state.train_metric_values[k].item() -
-                            self.train_prev_metric[k]
-                    })
+            for k in self.train_prev_metric.keys():
+                logger.log_metrics({
+                    f'metrics/train/{k}_fdiff':
+                        state.train_metric_values[k].item() -
+                        self.train_prev_metric[k]
+                })
 
             for k in state.train_metric_values.keys():
                 self.train_prev_metric[k] = state.train_metric_values[k].item()
 
     def eval_end(self, state: State, logger: Logger):
         if self.diff_eval_metrics:
-            if self.eval_prev_metric:
-                for k in state.eval_metric_values.keys():
+            evaluator = state.dataloader_label
+            metrics = list(state.eval_metrics[evaluator].keys())
+
+            for k in metrics:
+                mkey = '/'.join(['metrics', evaluator, k])
+                if mkey in self.eval_prev_metric.keys():
                     logger.log_metrics({
-                        f'metrics/eval/{k}_fdiff':
+                        f'{mkey}_fdiff':
                             state.eval_metric_values[k].item() -
-                            self.eval_prev_metric[k]
+                            self.eval_prev_metric[mkey]
                     })
 
             for k in state.eval_metric_values.keys():
```

Comment:
should this loop be over `metrics` instead of `state.eval_metric_values`? I think both will work (cause it will just overwrite with the same value a bunch of times), but the second is more correct?

Reply:
good catch.